core/src/main/java/org/elasticsearch/repositories/Repository.java

```diff
@@ -29,22 +33,19 @@
 
 import java.io.IOException;
 import java.util.List;
-import java.util.function.Predicate;
 
 /**
  * Snapshot repository interface.
  * <p>
- * Responsible for index and cluster level operations. It's called only on master.
- * Shard-level operations are performed using {@link org.elasticsearch.index.snapshots.IndexShardRepository}
- * interface on data nodes.
+ * Responsible for index and cluster and shard level operations.
  * <p>
  * Typical snapshot usage pattern:
  * <ul>
  * <li>Master calls {@link #initializeSnapshot(SnapshotId, List, org.elasticsearch.cluster.metadata.MetaData)}
  * with list of indices that will be included into the snapshot</li>
- * <li>Data nodes call {@link org.elasticsearch.index.snapshots.IndexShardRepository#snapshot(SnapshotId, ShardId, IndexCommit, IndexShardSnapshotStatus)} for each shard</li>
- * <li>When all shard calls return master calls {@link #finalizeSnapshot}
- * with possible list of failures</li>
+ * <li>Data nodes call {@link Repository#snapshot(IndexShard, SnapshotId, IndexCommit, IndexShardSnapshotStatus)}
```

Comment:
It'd be more helpful to me if these notes were on the method if we're going to mix them together like this.

Another option would be to have two interfaces implemented by one class. I'm not sure that helps at all.